### PR TITLE
Add document expiration feature. Fixes #1652.

### LINKED ISSUE #1652: Document expiration
Some standards bodies have a notion of documents which expire at some point. Internet-Drafts at the IETF work this way, for instance, and @WICG may add one (see WICG/admin#102).

It'd be nice if Bikeshed supported this or could automate this somehow. Here are some handwavy thoughts:

* It should be possible to do this by hand for an individual document via metadata. e.g. add Expires and have it take the same format as Date.
* It should be possible for document statuses to get this automatically by default.
* The default bikeshed boilerplate should take this into account and throw up a warning (warning-expired.include), and this should be overridable / customizable per-group and per-spec like all other boilerplate is today

### PR DESCRIPTION
**Pull request by hober**
Here's a first cut at an expiration feature like we're talking about in #1652.

One annoyance is this adds a dependency on `isodate` in order to parse ISO 8601 durations.

### CODE REVIEW

File: bikeshed/metadata.py [hunk 1]
```diff
@@ -334,6 +345,35 @@ def parseDate(key, val, lineNum):
         return None
 
 
+def parseDateOrDuration(key, val, lineNum):
```
Comment (tabatkins) [inline bikeshed/metadata.py:348]: Also accept "never", returning None, so you can override a default expiry. Maybe allow a boolish false too?

**Comment by tabatkins** [comment]
Add some tests and the one fix, and this is R+ from me

**Comment by tabatkins** [comment]
Well, I guess we'll want to do the "starts closed, uses js to open after expiration" behavior for the not-yet-expired case

File: bikeshed/metadata.py (hunk 1 above)
Comment (hober) [reply bikeshed/metadata.py:348]: Added never and boolish false in a33f871.

**Comment by hober** [comment]
> Add some tests and the one fix, and this is R+ from me

Added tests and accept never & boolish false in a33f871.

Haven't done the JS bit yet.

**Comment by tabatkins** [comment]
Okay, I added the JS and did some other minor fixup, including making sure the tests weren't dependent on the date they were generated, since that means they'll fail tomorrow. ^_^ Also added docs for the feature.

**Pull request merged**